technologies/job/knime/login-auth/instanceActions.js

```diff
@@ -0,0 +1,215 @@
+    }
+
+    return Response.success(JOB_STATES[data.state] || JobStatus.AWAITING);
+  } catch (error) {
+    if (error && error.response) {
+      return Response.error(`${ERRORS_MESSAGES.FAILED_TO_GET_STATUS_ERROR} : ${error.response.status} - ${error.response.statusText}`, { error: new Error(`${error.response.status} - ${error.response.statusText}`) });
+    }
+
+    return Response.error(ERRORS_MESSAGES.FAILED_TO_GET_STATUS_ERROR, { error });
+  }
+};
+
+/**
+ * Logic to retrieve the external job instance logs.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getLogs = async ({ job, instance }) => {
+  try {
+    console.log('GET LOG INSTANCE:', instance);
+    const result = await axios.get(
+      `${job.featuresValues.endpoint.url}/rest/v4/admin/logs`,
+      {
+        ...getRequestConfigFromEndpointForm(job.featuresValues.endpoint),
+        responseType: 'arraybuffer'
+      }
+    );
+
+    if (!result) {
+      return Response.error(ERRORS_MESSAGES.NO_RESPONSE_FROM_KNIME, { error: new Error(ERRORS_MESSAGES.NO_RESPONSE_FROM_KNIME) });
+    }
+
+    const jobLogsFilePath = '/tmp/knime-logs.zip';
+    const jobLogsFolderPath = '/tmp/knime-logs';
+
+    const { data } = result;
+
+    if (!data) {
+      return Response.error(ERRORS_MESSAGES.FAILED_TO_GET_LOGS_ERROR, { error: new Error(ERRORS_MESSAGES.FAILED_TO_GET_LOGS_ERROR) });
+    }
+
+    if (fs.existsSync(jobLogsFilePath)) {
+      fs.unlinkSync(jobLogsFilePath);
+    }
+
+    fs.appendFileSync(jobLogsFilePath, data);
+
+    if (fs.existsSync(jobLogsFolderPath)) {
+      rimraf.sync(jobLogsFolderPath);
+    }
+
+    await extract(jobLogsFilePath, { dir: jobLogsFolderPath });
+
+    const directories = fs.readdirSync(jobLogsFolderPath, { withFileTypes: true })
+      .filter(dirent => dirent.isDirectory())
+      .map(dirent => dirent.name);
+
+    const jobIdString = `: ${(instance && instance.payload && instance.payload.id) || job.featuresValues.job.id} :`;
+
+    let jobLogs = [];
+
+    directories.forEach((dir) => {
+      if (fs.existsSync(`${jobLogsFolderPath}/${dir}/knime.log`)) {
+
+        // We are getting all lines in new log file
+        const newLogs = fs.readFileSync(`${jobLogsFolderPath}/${dir}/knime.log`, 'utf8');
+
+        const logsLines = newLogs.split('\n');
+
+        if (logsLines && logsLines.length > 0) {
+          jobLogs = jobLogs.concat(getLogsForAJob(logsLines, 0, jobIdString));
+        }
+      }
+    });
+
+    return Response.success(jobLogs.map(
+      (jobLogLine) => {
+        const maybeDate = jobLogLine.substring(0, 19);
+        const isAValidDate = dayjs(maybeDate).isValid();
+        return Log(jobLogLine, Stream.STDOUT, isAValidDate && maybeDate);
```

Comment:
```suggestion
        return Log(jobLogLine, Stream.STDOUT, isAValidDate ? maybeDate : undefined);
```
The purpose is to avoid `false` as a value here.